core/src/main/scala/com/danielwestheide/kontextfrei/rdd/RDDBase.scala

```diff
@@ -1,7 +1,25 @@
 package com.danielwestheide.kontextfrei.rdd
 
 import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
 
 private[kontextfrei] trait RDDBase {
   def sparkContext: SparkContext
+
+  def withSite[A, R](as: RDD[A])(body: RDD[A] => R): R = {
```

Comment:
I am not sure whether the `RDD[A]` in needed as a parameter to the `withSite` function, because `RDDBase` already has access to the `SparkContext` via the `sparkContext` method, and the `RDD[A]` isn't used for anything else than getting the `SparkContext`.

Reply:
Yes you're correct it is redundant. However I was going to suggest a refactor (not yet sure exactly how) separating `DCollectionConstructors` from `DCollectionOps`, thus allowing the `val sparkContext` to move into `RDDConstructors`.
My motivation for such is that in my large production code base I have yet-to-be-migrated code using `RDD` directly that must interface to revised code using `DCollectionOps.` As things stand I find I need to create `implicit val sparkContext` in that older code so that the compiler can instantiate the `RDDOps` implicit for `DCollectionOps`. Not a big deal I admit but feels like it could be cleaner.
I'm open to removing the param for now if that keeps the code cleaner?

Reply:
I think the reasoning for your suggested refactoring makes a lot of sense. With that in mind, I think we can stick to the current implementation that requires to pass in the `RDD` to the `withSite` method.

BTW, it's great to hear that you are migrating to _kontextfrei_ in an existing production application. This is not only very motivating but also invaluable in terms of the feedback and suggestions for improvements I'm getting. :)